### PR TITLE
Add the ability to differential the display of credentials.

### PR DESCRIPTION
**Pull request by dirkcuys**
This is still a work in progress

- what credentials should be displayed differently?
- how should those credentials be displayed?
- any thoughts on how to check for type: `credential -> credentialSubject > hasCredential > type` contains an array with 'EducationalOccupationalCredential' and I assume whatever the specific credential is?
- any thoughts on how to organize and name components?

### CODE REVIEW

**Comment by dirkcuys** [comment]
@dmitrizagidulin @bmuramatsu this is where I got to with the custom display for student IDs:
Screenshot_1645194647

There are several unsettled issues with the display that might only be settled once we get requirements from a concrete use case. Eg.
- what is the correct schema for the student ID?
- are student images inline or links to web resources?
- the barcode is currently a 1px high image being stretched
- what do we do about the QR
- updates to the display

The second and maybe more pressing issue is that [app/screens/CredentialScreen/CredentialScreen.tsx](app/screens/CredentialScreen/CredentialScreen.tsx) displays the credential item in the list with a title equal to `credentialSubject.hasCredential.name`.

Screenshot_1645194656

If the wallet supports different credentials, we will need a way to derive a title for different types of credentials. Unless there is a simpler way to solve this, it would involve a similar approach of identifying the type of credential based on some logic and returning a title appropriate for each credential.

I left some pseudo code of how we could do this in https://github.com/dirkcuys/learner-credential-wallet/blob/credential-display/app/components/CredentialCard/CredentialCard.tsx#L18-L36. I'd probably structure the custom components a bit differently if going down this route so that the main thing being exported is that object defining `match`, `component` and `renderTitle`.

What are your thoughts.

**Comment by dirkcuys** [comment]
@dmitrizagidulin 

I've updated the way the CredentialCard determines what React component to use for display and added a way of determining the title for different credential types.

Essentially it boils down to a list of of functions that will return a dictionary with the component and title if it determines that the credential matches it's criteria:
```javascript
[ 
  (credential) => {
    if (!condition) return null;
    return {
        component: CustomCredentialDisplayComponent
        title: credential.field.that.contains.title
    }
  },
  ...
]
```
and a function that will call each function and return the result of the first one that returns something.

I'm wondering if there are other fields used where credentials are being displayed that might also present issues? Ex. 'app/components/CredentialItem/CredentialItem.tsx' takes both title, subtitle and image as properties. I've updated the title to use the method mentioned above.

ps. I see there is a conflict on the PR. I can fix it if we agree that the approach I took makes sense, but for testing I still need the debug code in settings and I think the conflict is because that code was removed.

**Comment by dirkcuys** [comment]
I've updated the student card component to use a single image rather than a separate barcode and picture. I also updated the type definitions to make the example student ID and degree completion credentials valid typescript.

Screenshot_1652362609

File: app/components/CredentialItem/CredentialItem.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import styles from './CredentialItem.styles';
 import type { CredentialItemProps } from './CredentialItem.d';
 
 export default function CredentialItem({
-  title,
+  title, // Should this info be passed in here, or determined by the CredentialItem?
```
Comment (dmitrizagidulin) [inline app/components/CredentialItem/CredentialItem.tsx:11]: Passed in I think.

**Review by dmitrizagidulin — APPROVED** [review]
:+1: 
Looks good. Let's resolve the merge conflicts, and we'll get it into the wallet.

**Pull request merged**